modAutomaticDialogPicker/content/scripts/local/automaticDialogPicker/random_dialog_picker.ws

```diff
@@ -14,6 +14,24 @@ statemachine class MCM_RandomDialogPicker {
     this.GotoState('Loading');
   }
 
+  function getBetterFlowChoiceWeight(choice: SSceneChoice): int {
```

Comment:
If i understand correctly, the function is called for every choice available in the current scene. That means that for scenes where none of the choices match, the function `GetLocStringById` will be called 4 times. And if we happen to use the system in other places that will scale even more.

As you may have seen for the shortcut choices i decided to store the strings in the class itself right after the game is loaded to prevent that from happening. It is not a perfect solution as it consumes more memory, but it has the advantage of not calling `GetLocStringById` X times every time dialogue choices are displayed.

From what i understand the `getBetterFlowChoiceWeight` is going to be called no matter what, so we could either pick the solution i went for the short circuit choices or we could pre-load those choices before the for loop?

Some pseudo code in case i did not express my pre-load idea well:
```ts

struct MCM_DialogFlowWeight {
  var description: string;
  var weight: int;
}

// ...

var dialog_flow_weights: array<MCM_DialogFlowWeight>;

// we could also imagine it pre-loads the combination of 4 choice weights only
// if the first one could be found in the `choices` list.
dialog_flow_weigths = = this.getDialogFlowWeights(choices);

for (i = 0; i < choices.Size(); i += 1) {
      choice = choices[i];

      // that would search if `choice` is in `dialog_flow_weigths` and returns the weight,
      //  it returns 0 otherwise
      weight = this.getBetterFlowChoiceWeight(dialog_flow_weigths , choice);

      // ...
```

What do you think?